prime-router/src/main/kotlin/history/db/ReportGraph.kt

```diff
@@ -408,15 +410,22 @@ class ReportGraph(
     private fun descendantReportRecords(
         txn: DataAccessTransaction,
         cte: CommonTableExpression<Record1<UUID>>,
-        searchedForTaskActions: Set<TaskAction>,
-    ) = DSL.using(txn)
-        .withRecursive(cte)
-        .select(REPORT_FILE.asterisk())
-        .distinctOn(REPORT_FILE.REPORT_ID)
-        .from(cte)
-        .join(REPORT_FILE)
-        .on(REPORT_FILE.REPORT_ID.eq(cte.field(0, UUID::class.java)))
-        .join(ACTION)
-        .on(ACTION.ACTION_ID.eq(REPORT_FILE.ACTION_ID))
-        .where(ACTION.ACTION_NAME.`in`(searchedForTaskActions))
+        searchedForTaskActions: Set<TaskAction>?,
+    ): SelectOnConditionStep<Record> {
+        val select = DSL.using(txn)
+            .withRecursive(cte)
+            .select(REPORT_FILE.asterisk())
+            .distinctOn(REPORT_FILE.REPORT_ID)
+            .from(cte)
+            .join(REPORT_FILE)
+            .on(REPORT_FILE.REPORT_ID.eq(cte.field(0, UUID::class.java)))
+            .join(ACTION)
+            .on(ACTION.ACTION_ID.eq(REPORT_FILE.ACTION_ID))
+
+        if (searchedForTaskActions != null) {
```

Comment:
In the case for the submission history we want to be look at the entire report graph

Reply:
wasn't the code that was there returning the entire report graph already? Assuming the caller used the endpoint given to them in response to a successful submission did not the legacy code return the descendants in order?

Reply:
It mostly did, but also didn't select out the ReportFile object.  Big picture trying to have the recursive queries be abstracted in one spot.